aicsimageio/tests/writers/test_ome_tiled_tiff_writer.py

```diff
@@ -0,0 +1,163 @@
+            (2, 2, 3, 4, 5, 3),
+            to_xml(OmeTiffWriter.build_ome([(2, 2, 3, 4, 5, 3)], [np.dtype(np.uint8)])),
+            "TCZYXS",
+            marks=pytest.mark.raises(exception=ValueError),
+        ),
+        pytest.param(
+            (2, 2, 3, 4, 5, 3),
+            OmeTiffWriter.build_ome([(2, 2, 3, 4, 5, 3)], [np.dtype(np.uint8)]),
+            "TCZYXS",
+            marks=pytest.mark.raises(exception=ValueError),
+        ),
+        # wrong dtype
+        pytest.param(
+            (1, 2, 3, 4, 5),
+            to_xml(OmeTiffWriter.build_ome([(1, 2, 3, 4, 5)], [np.dtype(np.float32)])),
+            "TCZYX",
+            marks=pytest.mark.raises(exception=ValueError),
+        ),
+        # wrong dtype
+        pytest.param(
+            (1, 2, 3, 4, 5),
+            OmeTiffWriter.build_ome([(1, 2, 3, 4, 5)], [np.dtype(np.float32)]),
+            "TCZYX",
+            marks=pytest.mark.raises(exception=ValueError),
+        ),
+        # wrong dims
+        pytest.param(
+            (1, 2, 3, 4, 5),
+            to_xml(OmeTiffWriter.build_ome([(2, 2, 3, 4, 5)], [np.dtype(np.float32)])),
+            "TCZYX",
+            marks=pytest.mark.raises(exception=ValueError),
+        ),
+        # wrong dims
+        pytest.param(
+            (1, 2, 3, 4, 5),
+            OmeTiffWriter.build_ome([(2, 2, 3, 4, 5)], [np.dtype(np.float32)]),
+            "TCZYX",
+            marks=pytest.mark.raises(exception=ValueError),
+        ),
+        # just totally wrong but valid ome
+        pytest.param(
+            (1, 2, 3, 4, 5),
+            to_xml(OME()),
+            "TCZYX",
+            marks=pytest.mark.raises(exception=IndexError),
+        ),
+        # just totally wrong but valid ome
+        pytest.param(
+            (1, 2, 3, 4, 5),
+            OME(),
+            "TCZYX",
+            marks=pytest.mark.raises(exception=IndexError),
+        ),
+        # even more blatantly bad ome
+        pytest.param(
+            (1, 2, 3, 4, 5),
+            "bad ome string",
+            "TCZYX",
+            # raised from within ome-types
+            marks=pytest.mark.raises(exception=urllib.error.URLError),
+        ),
+    ],
+)
+@pytest.mark.parametrize("filename", ["e.ome.tiff"])
+def test_ome_tiff_writer_with_meta(
+    array_constructor: Callable,
+    shape_to_create: Tuple[int, ...],
+    ome_xml: Union[str, OME, None],
+    expected_dim_order: Tuple[int, ...],
+    filename: str,
+) -> None:
+    # Create array
+    arr = array_constructor(shape_to_create, dtype=np.uint8)
+
+    # Construct save end point
+    save_uri = get_resource_write_full_path(filename, LOCAL)
+
+    # Normal save
+    OmeTiledTiffWriter.save(arr, save_uri, dimension_order=None, ome_xml=ome_xml)
+
+    # Read written result and check basics
+    reader = OmeTiffReader(save_uri)
+
+    # Check basics
+    assert len(reader.scenes) == 1
+    assert reader.shape == shape_to_create
+    assert reader.dims.order == expected_dim_order
```

Comment:
Love all of these tests. Only request is to add a two other tests (and maybe they are the same).

1. a test for multiple tiles. If the default tile size is 1024 then can you write out an OME TIFF with 4 tiles or similar?
2. a test for writing out dask array with multiple tiles. Not sure how to specifically test that it is only computing one tile in memory at a time though...

If 2 isn't possible then please at least do 1.

Reply:
2 should be possible. However, for now we just write plane by plane, rather than tile by tile. The main reason is performance. It looks like most of the readers load an entire plane regardless of what the size of the subregion within a plane that is requested.

Reply:
Ahhhh. I couldn't tell how much data was being pulled into memory at a single time in the dask array case. Plane by plane is still better than the whole thing into memory for sure. Still not sure how to test that haha

Reply:
Ah...right good point. We might be able to do this by constricting the available memory utilization in the test using memray. That of course uses a new dependency. We are looking at using this for testing in bfio with a new set of unit tests.

https://pypi.org/project/pytest-memray/

Reply:
I am fine with adding a pytest plugin to the test deps. No worries from me there.